Src/ScipBe.Common.Office/OneNote/OneNoteProvider.cs

```diff
@@ -73,61 +73,62 @@ public static IEnumerable<IOneNoteExtPage> FindPages(string searchString)
         {
             var section = new OneNoteExtSection()
             {
-                ID = element.Attribute("ID").Value,
-                Name = element.Attribute("name").Value,
-                Path = element.Attribute("path").Value,
-                Color = element.Attribute("color").Value != "none" ? ColorTranslator.FromHtml(element.Attribute("color").Value) : (Color?)null,
-                Encrypted = (element.Attribute("encrypted") != null) && (element.Attribute("encrypted").Value == "true"),
+                ID = sectionElement.Attribute("ID").Value,
+                Name = sectionElement.Attribute("name").Value,
+                Path = sectionElement.Attribute("path").Value,
+                Color = sectionElement.Attribute("color").Value != "none" ? ColorTranslator.FromHtml(sectionElement.Attribute("color").Value) : (Color?)null,
+                Encrypted = (sectionElement.Attribute("encrypted") != null) && (sectionElement.Attribute("encrypted").Value == "true"),
             };
 
             if (addPages)
             {
-                section.Pages = element.Elements(oneNamespace + "Page").Select(p => ParsePage(p, oneNamespace, false));
+                section.Pages = sectionElement.Elements(oneNamespace + "Page").Select(p => ParsePage(p, oneNamespace, false));
             }
 
             return section;
         }
 
-        private static IOneNoteExtPage ParsePage(XElement element, XNamespace oneNamespace, bool addParents)
+        private static IOneNoteExtPage ParsePage(XElement pageElement, XNamespace oneNamespace, bool addParents)
         {
             var page = new OneNoteExtPage()
             {
-                ID = element.Attribute("ID").Value,
-                Name = element.Attribute("name").Value,
-                Level = element.Attribute("pageLevel").Value.ToInt32(),
-                DateTime = element.Attribute("dateTime").Value.ToString().ToDateTime(),
-                LastModified = element.Attribute("lastModifiedTime").Value.ToString().ToDateTime(),
+                ID = pageElement.Attribute("ID").Value,
+                Name = pageElement.Attribute("name").Value,
+                Level = pageElement.Attribute("pageLevel").Value.ToInt32(),
+                DateTime = pageElement.Attribute("dateTime").Value.ToString().ToDateTime(),
+                LastModified = pageElement.Attribute("lastModifiedTime").Value.ToString().ToDateTime(),
             };
 
             if (addParents)
             {
-                page.Section = ParseSection(element.Parent, oneNamespace, false);
+                var sectionElement = pageElement.Parent;
+                page.Section = ParseSection(sectionElement, oneNamespace, false);
 
-                var notebookElement = element.Parent.Parent;
-                if (notebookElement.Name.LocalName == "SectionGroup")
+                var notebookElement = sectionElement.Parent;
+                while (notebookElement.Name.LocalName == "SectionGroup")
```

Comment:
This right here is the real fix. Previously I was only handling 1 level of nesting (Notebook>Section Group>Section), but there can be N nested section groups, so this takes care of that